### ISSUE TITLE
Bound effects for brightness and contrast attacks

### ISSUE THREAD

**Issue by Ne-ki-tos** [AI-secure/semantic-randomized-smoothing#7]
Hello, thanks for releasing the code.
I have a question on how you certify against brightness and contrast change attacks. I've read your paper and I believe you do not say anything about the fact that addition and multiplication of pixel values can possibly get us out of [0,1] range. Is it true or am I missing something? It seems that in your code you also do not clip pixel values to [0,1] after applying brightness/contrast attacks. Thus, the resulting attacks differ from realistic ones which cannot take pixel values out of [0,1] bounds. Please, tell me whether my concerns are valid.

**Comment by llylly** [comment]
Thanks for your interest in our work!
Throughout the paper, we didn't assume the image input domain is [0,1]^d. Instead, it is R^d.
Note that this threat model corresponds to the case where the image signal from the camera is directly fed into the neural network model **without** clipping.
To handle the clipping, one possible extension is applying (clip(x, a) o Brightness(b) o contract(c)) as the smoothing distribution, where a, b, c are Gaussian distributed scalars as smoothing variables. In this case, you can still leverage TSS-R to compute the certification.

**Comment by Ne-ki-tos** [comment]
Thank you for clearing that to me.
If your domain is R^d and you do not apply clipping, is it correct to compare your results with those from Mohapatra et al. “Towards Verifying Robustness of Neural Networks Against A Family of Semantic Perturbations” who use clipping? You do that in paper but now it seems that attack types are different.

**Comment by llylly** [comment]
Thank you for pointing out this detail! We found that only Mohapatra et al. use this threat model (change + clipping) while the other two baselines that consider brightness and contrast ("Certifying Geometric Robustness of Neural Networks" and "An Abstract Domain for Certifying Neural Networks") also do not consider clipping.
To reflect this difference in the latent threat models, we have updated our paper to explicitly mention this, and the new version will be released on ArXiv this week.
